src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -693,6 +694,17 @@ private void RegisterApplicationLifetimeEvents()
             _applicationLifetime.ApplicationStopping.Register(() =>
             {
                 Interlocked.Exchange(ref _applicationStopping, 1);
+                if (_environment.IsDrainOnApplicationStopping())
+                {
+                    var drainModeManager = _host?.Services.GetService<IDrainModeManager>();
+                    if (drainModeManager != null)
+                    {
+                        _logger.LogDebug("Application Stopping: enabling drain mode");
```

Comment:
I don't have context on the reason for this change - I didn't see a linked issue or anything? When the application is stopping, all hosted services are also stopping, which will stop the ScriptHost, which will stop listeners and do regular shutdown, etc. already naturally as we've always done. Is there some problem we've identified with that existing behavior?

In WebJobs outside of functions, a customer will similarly call JobHost.StopAsync to shut things down, and they don't know anything about or use IDrainModeManager.

Reply:
@mathewc this PR is related to the sync up we had on waiting on all inprogress function executions before shutting down the host. In Kubernetes that's orchestrated using SIGTERM (ApplicationStopping event in .NET)